pkg/kn/commands/revision/list.go

```diff
@@ -85,6 +86,19 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
 				}
 			}
 
+			// Add revision service info as annocations: traffic and tags
+			for _, revision := range revisionList.Items {
+				serviceName := revision.Labels[serving.ServiceLabelKey]
+				service, err := client.GetService(serviceName)
```

Comment:
Can we please cache the service and don't refetch afresh for every revision (belonging to the same service) ?

Reply:
Fair. Will do.